hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -92,13 +101,42 @@ public void associateWith(List<Token> tokens, int maxAllowed) {
 		associatedTokens.addAllIds(uniqueIds);
 	}
 
-	public void dissociateWith(List<Token> tokens) {
+	public void dissociateWith(List<Pair<TokenRelationship, TokenRelationship>> relations, final OptionValidator validator) {
 		final Set<Id> uniqueIds = new HashSet<>();
-		for (var token : tokens) {
+		for (var rel : relations) {
+			/*	extraction	 */
+			final var accountRel = rel.getKey();
+			final var treasuryRel = rel.getValue();
+			final var account = accountRel.getAccount();
+			final var treasury = treasuryRel.getAccount();
+			final var token = treasuryRel.getToken();
+			/*	validation	 */
+			validateTrue(!token.isDeleted(), TOKEN_WAS_DELETED, () ->
```

Comment:
If you go through the previous `disassociate` implementation you can see that if a token is deleted, an account can still do token disassociate in order to release its relationship with the token

Reply:
So - we will validate that the **account is not treasury** only when the token is **not** deleted?